hbase-server/src/test/java/org/apache/hadoop/hbase/master/assignment/TestRegionStateStore.java

```diff
@@ -84,6 +89,36 @@ public void visitRegionState(Result result, RegionInfo regionInfo, RegionState.S
     assertTrue("Visitor has not been called.", visitorCalled.get());
   }
 
+  @Test
+  public void testVisitMetaForBadRegionState() throws Exception {
+    final TableName tableName = TableName.valueOf("testVisitMetaForBadRegionState");
+    util.createTable(tableName, "cf");
+    final List<HRegion> regions = util.getHBaseCluster().getRegions(tableName);
+    final String encodedName = regions.get(0).getRegionInfo().getEncodedName();
+    final RegionStateStore regionStateStore = util.getHBaseCluster().getMaster().
+        getAssignmentManager().getRegionStateStore();
+
+    // add the BAD_STATE which does not exist in enum RegionState.State
+    Put put = new Put(regions.get(0).getRegionInfo().getRegionName(), EnvironmentEdgeManager.currentTime());
+    put.addColumn(HConstants.CATALOG_FAMILY, HConstants.STATE_QUALIFIER,
+        Bytes.toBytes("BAD_STATE"));
+
+    try (Table table = util.getConnection().getTable(TableName.META_TABLE_NAME)) {
+      table.put(put);
+    }
+
+    final AtomicBoolean visitorCalled = new AtomicBoolean(false);
+    regionStateStore.visitMetaForRegion(encodedName, new RegionStateStore.RegionStateVisitor() {
+      @Override
+      public void visitRegionState(Result result, RegionInfo regionInfo, RegionState.State state,
+                                   ServerName regionLocation, ServerName lastHost, long openSeqNum) {
+        assertEquals(encodedName, regionInfo.getEncodedName());
```

Comment:
Need to check if the state is null

Reply:
Agree with @guangxuCheng comments above.

Reply:
@guangxuCheng @wchevreuil  done, thanks for pointing it out.